app/models/katello/upstream_consumer.rb

```diff
@@ -0,0 +1,21 @@
+require 'katello/resources/candlepin'
+
+module Katello
+  class UpstreamConsumer < OpenStruct
+    class << self
+      def bind_entitlements(organization, pools: [])
+        map_pool_attributes(pools) do |pool|
+          Resources::Candlepin::UpstreamConsumer.bind_entitlements(organization, pool: pool)
```

Comment:
It looks like this is issuing a bind request for each pool given since we are yielding for each one - but did you know that you can send a body with your request upstream much like we are taking input to your API:

```
[
  {
    "poolId": "string",
    "quantity": 0
  }
]
```

This way we can attach in bulk. Not only is this faster, but if something fails the entire transaction will be rolled back (if I remember correctly) and so the user won't be left in some in-between state. I think we should use that feature!

Reply:
Yeah, I noticed that in the docs, but I was unable to get it to work. I will try again, but perhaps you could verify that endpoint works for you.

Reply:
Will check :eye: